CMP1903_A1_2324/Die.cs

```diff
@@ -3,21 +3,34 @@
 using System.Linq;
 using System.Text;
 using System.Threading.Tasks;
+using System.Diagnostics;
 
 namespace CMP1903_A1_2324
 {
     internal class Die
-    {
-        /*
-         * The Die class should contain one property to hold the current die value,
-         * and one method that rolls the die, returns and integer and takes no parameters.
-         */
-
-        //Property
 
+    {
+        private int currentValue;
+        private static Random random;
 
-        //Method
+        // The Die constructor initialises the random number generator. Each new instance of the die class will be generated with its own independent "random".
+        public Die()
+        {
+            random = new Random();
+        }
 
+        // Method that rolls the die and return the result
+        public int Roll()
+        {
+            // Generates a random number between 1 and 6. Next is a method of the Random class in .Net
+            currentValue = random.Next(1, 7); // Numbers generated are stored in the currentValue variable.
+            return currentValue;
+        }
 
+        // Gets the current value of the die
+        public int CurrentValue
```

Comment:
I would also add a set method to show encapsulation

Reply:
I have add "set { _currentValue = value; }" to demonstrate this